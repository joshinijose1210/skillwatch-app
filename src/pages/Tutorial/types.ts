export interface TutorialItem {
    videoId: string;
    cardLabel: string;
    cardDescription: string;
    category: string;
    cardTag: string[];
    thumbnail: string;
}

export interface FlexBoxRowProps {
    gap?: string;
}

export interface TagChipProps {
    textColor?: string;
}
