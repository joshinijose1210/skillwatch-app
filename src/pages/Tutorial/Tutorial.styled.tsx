import { Card } from '@medly-components/core';
import styled from 'styled-components';
import { FlexBoxRowProps } from './types';

export const StyledCard = styled(Card)`
    min-width: 35rem;
    max-width: 32%;
    height: 60rem;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.12);
    gap: 1rem;
    /* margin: 2rem 0; */
    overflow: hidden;

    @media screen and (width>=1600px) {
        max-width: 23%;
    }
`;

export const CardsWrapper = styled.div`
    display: flex;
    flex-wrap: wrap;
    margin: 3rem 0;
    column-gap: 1.5rem;
    row-gap: 3rem;

    @media screen and (width>=1800px) {
        gap: 3rem;
    }
`;

export const FlexBoxRow = styled.div<FlexBoxRowProps>`
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    gap: ${({ gap }) => gap || '1rem'};
    align-items: center;
`;
