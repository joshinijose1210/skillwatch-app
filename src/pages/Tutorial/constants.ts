import { TutorialItem } from './types';

export const FILTER_OPTIONS = [
    { label: 'All', value: 'all' },
    { label: 'Setup', value: 'setup' },
    { label: '360-degree Feedback', value: '360-degree-feedback' },
    { label: 'KRA and KPI Management', value: 'KRA-and-KPI-Management' },
    { label: 'Performance Review', value: 'performance-review' },
    { label: 'Reports & Analytics', value: 'reports-and-analytics' }
];

const path = process.env.MODE === 'production' ? '/app' : '';

// Tutorial data source (can be replaced with dynamic API logic later)
export const TUTORIALS_DATA: TutorialItem[] = [
    {
        videoId: 'nfDYeOgeulI',
        cardLabel: 'Getting Started with SkillWatch Setup',
        cardDescription:
            'A simple guide to setting up your company, departments, teams, designations, roles & permissions, and employees — all in one smooth flow.',
        category: 'Setup',
        cardTag: ['Company', 'Department', 'Team', 'Designation', 'Roles & Permissions', 'Employees'],
        thumbnail: `${path}/thumbnails/tutorial1.jpg`
    },
    {
        videoId: 'VIY13-DyTZs',
        cardLabel: 'Giving Feedback the Right Way',
        category: '360-degree Feedback',
        cardDescription: `Whether it's positive feedback, improvement pointers, or appreciation notes — learn how to leave impactful feedback that truly helps.`,
        cardTag: ['Peer Feedback', 'Add Feedback', 'View Feedback', 'Manager Feedback', 'Performance Review'],
        thumbnail: `${path}/thumbnails/tutorial2.jpg`
    },
    {
        videoId: 'IML1FmcxQG8',
        cardLabel: 'Mastering the Art of Requesting Feedback',
        category: '360-degree Feedback',
        cardDescription: `Learn how to collect well-rounded feedback from team members, managers & clients — build a culture of growth, clarity, and continuous improvement.`,
        cardTag: ['Request Feedback', 'Multi-Person Feedback', 'Client/Stakeholder Feedback', 'Performance Review'],
        thumbnail: `${path}/thumbnails/tutorial3.jpg`
    },
    {
        videoId: 'uJUdECLBXSs',
        cardLabel: 'Slack Integration to Power Up SkillWatch!',
        cardDescription: `Plug into Slack to send feedback, request reviews, and get instant alerts for appreciation, suggestions, and performance review cycle — right where your team chats!`,
        category: 'Setup',
        cardTag: ['Slack', 'Integration', 'Add Feedback', 'Request Feedback', 'Reminders', 'Performance Review', 'Suggestion'],
        thumbnail: `${path}/thumbnails/tutorial4.jpg`
    },
    {
        videoId: 'aBibLJwCt3E',
        cardLabel: 'SkillWatch Features inside Slack: Feedback Made Effortless',
        cardDescription:
            'Discover how Slack + SkillWatch keeps feedback, reminders, and performance review updates at your fingertips — fast, easy, and integrated. All without leaving your slack chat!',
        category: '360-degree Feedback',
        cardTag: ['Add Feedback', 'Request Feedback', 'Multi-Person Feedback', 'Slack', 'Integration'],
        thumbnail: `${path}/thumbnails/tutorial5.jpg`
    },
    {
        videoId: '5CzV_KpmmhQ',
        cardLabel: 'Set KRA Weightage: Align Performance with Purpose',
        cardDescription:
            'A quick walkthrough on setting balanced weightages across KRAs (Knowledge, Results, and Attitude) to reflect your company’s expectations and ensure fair performance reviews.',
        category: 'Setup',
        cardTag: ['KRAs', 'KRA Weightage', 'Responsibilities', 'Performance Review'],
        thumbnail: `${path}/thumbnails/tutorial6.jpg`
    },
    {
        videoId: 'FdlyLney30c',
        cardLabel: 'Review Cycle Setup: From Kickoff to Check-In',
        cardDescription:
            'Learn how to define your performance review cycle and break it down into structured phases for self-reviews, manager reviews, and check-in with managers — all while staying on schedule and avoiding last minute chaos.',
        category: 'Setup',
        cardTag: ['Set Review Cycle', 'Review Timeline', 'Self Review', 'Manager Review', 'Check-in with Manager', 'Performance Review'],
        thumbnail: `${path}/thumbnails/tutorial7.jpg`
    }

    // {
    //     videoId: 'nfDYeOgeulI',
    //     cardLabel: 'Review Cycle Setup: From Kickoff to Check-In',
    //     cardDescription:
    //         'Learn how to define your performance review cycle and break it down into structured phases for self-reviews, manager reviews, and check-in with managers — all while staying on schedule and avoiding last-minute chaos.',
    //     category: 'Performance Review',
    //     cardTag: ['Review Timeline', 'Self Review', 'Manager Review', 'Check-in with Manager', 'Performance Review'],
    //     thumbnail: '/thumbnails/tutorial1.jpg'
    // }
];

export const categoryColorMap: {
    [key: string]: { bg: string; text: string };
} = {
    Setup: {
        bg: '#eff9ff',
        text: '#3872d2'
    },
    '360-degree Feedback': {
        bg: '#F2A14226',
        text: '#C0741A'
    },
    Integration: {
        bg: 'F2A14226',
        text: '#C0741A'
    },
    'KRA and KPI Management': {
        bg: '#5CBE4C26',
        text: '#29A016'
    },
    'Performance Review': {
        bg: '#4ACEBE26',
        text: '#0D9887'
    }
};
