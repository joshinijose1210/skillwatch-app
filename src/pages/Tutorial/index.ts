export { Tutorial as default } from './Tutorial';
