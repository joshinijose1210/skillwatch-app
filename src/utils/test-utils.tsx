import { ToastContainer } from '@medly-components/core';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { store as rtkStore, setupStore } from '@slice';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { defaultTheme } from '@theme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import reduxMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { userMockState } from '../mocks/userMockState';
import { initialState } from '../slice';

interface CustomOptions extends RenderOptions {
    initialState?: any;
    initialEntries?: any;
}

interface WithThemeProviderProps {
    children?: React.ReactNode;
}

const mockStore = reduxMockStore(),
    store = mockStore({
        ...initialState,
        user: { ...initialState.user, groups: ['admin'] }
    });

const WithThemeProvider: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <ThemeProvider theme={defaultTheme}>
        <>
            <ToastContainer position="top-end" />
            {children}
        </>
    </ThemeProvider>
);

const WithStore: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <WithThemeProvider>{children}</WithThemeProvider>
    </Provider>
);

const WithRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <MemoryRouter>
        <WithThemeProvider>{children}</WithThemeProvider>
    </MemoryRouter>
);

const WithStoreAndRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <MemoryRouter>
            <WithThemeProvider>{children}</WithThemeProvider>
        </MemoryRouter>
    </Provider>
);

export const RTKRender = (ui: React.ReactElement, { initialState = userMockState, ...renderOptions }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);

    const RtkWrapper: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
        <Provider store={store}>
            <MemoryRouter initialEntries={renderOptions.initialEntries}>
                <WithThemeProvider>{children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
    return { rtkStore, ...render(ui, { wrapper: RtkWrapper, ...renderOptions }) };
};
const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithThemeProvider, ...options });

export const renderWithStore = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStore, ...options });

export const renderWithRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithRouter, ...options });

export const renderWithStoreAndRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStoreAndRouter, ...options });

export const wrapper = (props: { children: React.ReactNode }, { initialState = userMockState }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);
    return (
        <Provider store={store}>
            <MemoryRouter>
                <WithThemeProvider>{props.children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render, store };
