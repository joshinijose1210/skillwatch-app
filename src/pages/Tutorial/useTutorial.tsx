import { useEffect, useMemo, useState } from 'react';
import { FILTER_OPTIONS, TUTORIALS_DATA } from './constants';

const API_KEY = process.env.YOUTUBE_API_KEY;

async function getYoutubeViewCount(videoId: string) {
    const response = await fetch(`https://www.googleapis.com/youtube/v3/videos?part=statistics&id=${videoId}&key=${API_KEY}`);
    const data = await response.json();
    const viewCount = data.items[0]?.statistics?.viewCount;
    return viewCount;
}

export const useTutorial = (videoId?: string) => {
    const [filter, setFilter] = useState<string>('all');
    const [searchText, setSearchText] = useState<string>('');
    const [viewCount, setViewCount] = useState<number>(0);
    const [openModal, setOpenModal] = useState<boolean>(false);

    const toggleModal = () => setOpenModal(prev => !prev);

    const onSearchChange = (text: string) => setSearchText(text.toLowerCase());

    const onSearchClear = () => setSearchText('');

    useEffect(() => {
        if (!videoId) return;
        (async () => {
            try {
                const views = await getYoutubeViewCount(videoId);
                setViewCount(views);
            } catch (error) {
                console.error('Error fetching view count:', error);
            }
        })();
    }, [videoId]);

    const filteredTutorials = useMemo(() => {
        return TUTORIALS_DATA.filter(tutorial => {
            const matchesFilter = filter === 'all' || tutorial.category === FILTER_OPTIONS.find(opt => opt.value === filter)?.label;

            const matchesSearch =
                tutorial.cardLabel.toLowerCase().includes(searchText) ||
                tutorial.cardDescription.toLowerCase().includes(searchText) ||
                tutorial.cardTag.some(tag => tag.toLowerCase().includes(searchText)) ||
                tutorial.category.toLowerCase().includes(searchText);
            return matchesFilter && matchesSearch;
        });
    }, [searchText, filter]);

    return {
        searchText,
        viewCount,
        filter,
        onSearchChange,
        onSearchClear,
        setFilter,
        openModal,
        toggleModal,
        filteredTutorials
    };
};
