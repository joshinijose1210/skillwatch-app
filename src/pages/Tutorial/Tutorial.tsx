import { StyledEmptyText, StyledSearchBox, StyledSingleSelect } from '@common';
import { PageContent } from '@components';
import TutorialCardModal from '@components/TutorialCardModal/TutorialCardModal';
import ListHeader from '@components/reusableComponents/ListHeader';
import { Text } from '@medly-components/core';
import { CardsWrapper, FlexBoxRow } from './Tutorial.styled';
import { FILTER_OPTIONS } from './constants';
import { useTutorial } from './useTutorial';

export const Tutorial = () => {
    const { searchText, filter, setFilter, onSearchChange, onSearchClear, filteredTutorials } = useTutorial();

    return (
        <PageContent>
            <ListHeader
                title={
                    <Text textVariant="h3" style={{ fontSize: '20px' }} textWeight="Medium">
                        Tutorial Videos
                    </Text>
                }
            />

            <FlexBoxRow>
                <StyledSearchBox
                    placeholder="Search by title, description, or tag"
                    onInputChange={onSearchChange}
                    onClear={onSearchClear}
                    value={searchText}
                    size="M"
                    className="h-40"
                    minWidth="30rem"
                    data-testid="search-tutorial"
                />

                <StyledSingleSelect
                    options={FILTER_OPTIONS}
                    variant="outlined"
                    size="S"
                    placeholder="Select Category"
                    label="Select Category"
                    className="h-40"
                    value={filter}
                    isSearchable
                    minWidth="30rem"
                    onChange={val => val && setFilter(val)}
                    data-testid="filter"
                />
            </FlexBoxRow>

            <CardsWrapper>
                {filteredTutorials.length > 0 ? (
                    filteredTutorials.map((tutorial, index) => <TutorialCardModal key={index} video={tutorial} />)
                ) : (
                    <StyledEmptyText style={{ marginTop: '2rem' }}>More tutorial videos coming soon...</StyledEmptyText>
                )}
            </CardsWrapper>
        </PageContent>
    );
};
