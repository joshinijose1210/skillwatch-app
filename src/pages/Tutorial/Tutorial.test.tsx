import { fireEvent, queryByText, screen, waitFor } from '@testing-library/react';
import { RTKRender } from '@utils/test-utils';
import { Tutorial } from './Tutorial';
import { useTutorial } from './useTutorial';

jest.mock('./useTutorial');

const mockSetFilter = jest.fn();
const mockOnSearchChange = jest.fn();
const mockOnSearchClear = jest.fn();

beforeAll(() => {
    Element.prototype.scrollIntoView = jest.fn();
});

const mockTutorials = [
    {
        videoId: 'nfDYeOgeulI',
        cardLabel: 'Getting Started with SkillWatch Setup',
        cardDescription: 'A simple guide to setting up your company',
        category: 'Setup',
        cardTag: ['Company'],
        thumbnail: '/thumbnails/tutorial1.jpg'
    },
    {
        videoId: 'efgh1234',
        cardLabel: 'Giving Feedback the Right Way',
        cardDescription: 'Leave impactful feedback',
        category: '360-degree Feedback',
        cardTag: ['Peer Feedback'],
        thumbnail: '/thumbnails/tutorial1.jpg'
    }
];

const setupTutorialHook = (tutorials = mockTutorials, searchText = '', filter = 'all') => {
    (useTutorial as jest.Mock).mockReturnValue({
        searchText,
        filter,
        viewCount: 100,
        setFilter: mockSetFilter,
        onSearchChange: mockOnSearchChange,
        onSearchClear: mockOnSearchClear,
        openModal: false,
        toggleModal: jest.fn(),
        filteredTutorials: tutorials
    });
};

const renderTutorial = () => RTKRender(<Tutorial />);

describe('Tutorial Component', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        setupTutorialHook();
    });

    it('should take snapshot', () => {
        const { container } = renderTutorial();
        expect(container).toMatchSnapshot();
    });

    it('should render header and controls', () => {
        renderTutorial();

        expect(screen.getByText('Tutorial Videos')).toBeInTheDocument();
        expect(screen.getByTestId('search-tutorial')).toBeInTheDocument();
        expect(screen.getByTestId('filter')).toBeInTheDocument();
    });

    it('should render all tutorial cards', () => {
        renderTutorial();

        expect(screen.getByText('Getting Started with SkillWatch Setup')).toBeInTheDocument();
        expect(screen.getByText('Giving Feedback the Right Way')).toBeInTheDocument();
    });

    it('should show empty state if no tutorials are returned', () => {
        setupTutorialHook([]);
        renderTutorial();

        expect(screen.getByText('More tutorial videos coming soon...')).toBeInTheDocument();
    });

    it('should trigger onSearchChange on input', () => {
        renderTutorial();

        const searchBox = screen.getByTestId('search-tutorial');
        fireEvent.change(searchBox, { target: { value: 'Feedback' } });

        expect(mockOnSearchChange).toHaveBeenCalledWith('Feedback');
    });

    // TODO: to be fixed later
    // it('should call setFilter when "Setup" is selected', async () => {
    //     renderTutorial();

    //     const filterDropdown = screen.getByTestId('filter');
    //     fireEvent.change(filterDropdown, { target: { value: 'setup' } });
    //     screen.debug();
    //     expect(screen.queryByText('360-degree Feedback')).not.toBeInTheDocument();
    // });
});
