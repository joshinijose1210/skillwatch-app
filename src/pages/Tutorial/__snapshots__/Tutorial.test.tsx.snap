// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tutorial Component should take snapshot 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          <h3
            class="sc-1ntgl3l-0 kcLBOi"
            textalign="initial"
            textcolor="inherit"
            textvariant="h3"
            textweight="Medium"
          >
            Tutorial Videos
          </h3>
        </h3>
      </div>
    </div>
    <div
      class="sc-groybc-2 gxbRAq"
    >
      <div
        class="sc-1cnun24-0 kEKbcv sc-tkiih6-15 cvCOyL"
        maxwidth="100%"
        minwidth="30rem"
      >
        <input
          class="sc-9l9lu7-0 lfgaXg"
          data-testid="search-tutorial"
          placeholder="Search by title, description, or tag"
          value=""
        />
        <span
          class="sc-1lts63a-1 lckNMu"
        >
          <svg
            class="sc-1pf3x8a-0 ivKmVS"
            fill="none"
            height="1em"
            title="search icon"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M9.063 3.125a5.938 5.938 0 1 0 0 11.875 5.938 5.938 0 0 0 0-11.875zM1.875 9.063a7.187 7.187 0 1 1 12.692 4.621l3.375 3.374a.625.625 0 1 1-.884.884l-3.374-3.374A7.188 7.188 0 0 1 1.875 9.063z"
              fill="#606878"
              fill-rule="evenodd"
              stroke="#606878"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width=".3"
            />
          </svg>
        </span>
      </div>
      <div
        class="sc-1hcsztp-0 DMwPt sc-tkiih6-13 cecHDR"
        id="filter-wrapper"
        minwidth="30rem"
        variant="outlined"
      >
        <div
          class="sc-17d4s39-0 ihsYRx"
          id="filter-input-wrapper"
          minwidth="30rem"
        >
          <div
            class="sc-19px6zp-0 jVlHWE"
            filled="[object Object]"
            fusion="[object Object]"
            minrows="1"
            outlined="[object Object]"
            textvariant="[object Object]"
            variant="outlined"
          >
            <div
              class="sc-lhr232-0 btNOKw"
              variant="outlined"
            >
              <input
                aria-describedby="filter-helper-text"
                autocomplete="off"
                class="sc-hb99wz-0 fnNpdL"
                data-testid="filter"
                errortext=""
                id="filter-input"
                inputsize="S"
                placeholder="Filter"
                type="text"
                value="All"
                variant="outlined"
              />
              <label
                class="sc-1du51u0-0 hWhrlz"
                for="filter-input"
                inputwidth="0"
                variant="outlined"
              >
                Filter
              </label>
            </div>
            <svg
              class="sc-1pf3x8a-0 bMQNVa sc-dynlvq-0 bdlxlZ"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                fill="#000"
              />
            </svg>
          </div>
          
        </div>
      </div>
    </div>
    <div
      class="sc-groybc-1 kSjrWK"
    >
      <div
        alignitems="left"
        backgroundcolor="#ffffff"
        borderradius="0.8rem"
        class="sc-gr411i-0 XbrlJ sc-groybc-0 iOwCrz"
        display="flex"
        flex="1"
        flowdirection="vertical"
        padding="[object Object]"
        separatorcolor="#98A7B7"
        shadowcolor="#b0bcc8"
        variant="solid"
      >
        <div
          class="sc-ysclju-4 cnWXyj"
        >
          <img
            alt="video"
            class="sc-ysclju-5 hJFxSB"
            src="/thumbnails/tutorial1.jpg"
          />
          <div
            class="sc-ysclju-6 mPJQr"
          />
          <div
            class="sc-ysclju-7 hLjIhT"
          >
            <svg
              class="sc-1pf3x8a-0 iEHpIU"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.927 6.82v10.36c0 .79.87 1.27 1.54.84l8.14-5.18a1 1 0 0 0 0-1.69l-8.14-5.17a.998.998 0 0 0-1.54.84z"
                fill="#000"
                fill-opacity=".54"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-ysclju-0 eRdnVw"
        >
          <h4
            class="sc-1ntgl3l-0 jPeJxZ sc-ysclju-8 jddxpz"
            data-tooltip-id="Getting Started with SkillWatch Setup-card-label"
            textalign="initial"
            textcolor="inherit"
            textvariant="h4"
            textweight="Medium"
          >
            Getting Started with SkillWatch Setup
          </h4>
          <button
            class="sc-1qiq5dc-0 bBZDuT sc-ysclju-10 wdhtS"
            color="#eff9ff"
            textcolor="#3872d2"
            variant="solid"
          >
            <span
              class="sc-1ntgl3l-0 hGagSU"
              textalign="initial"
              textcolor="inherit"
            >
              Setup
            </span>
          </button>
          <span
            class="sc-1ntgl3l-0 hGagSU sc-ysclju-9 VkrDH"
            data-tooltip-id="Getting Started with SkillWatch Setup-card-description"
            textalign="initial"
            textcolor="inherit"
            textweight="Regular"
          >
            A simple guide to setting up your company
          </span>
          <div
            class="sc-groybc-2 fQSnmt"
            gap="0"
          >
            <button
              class="sc-1qiq5dc-0 hmnFUL sc-ysclju-10 bmgBhz"
              color="#F3F4F6"
              variant="solid"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU"
                textalign="initial"
                textcolor="inherit"
              >
                #Company
              </span>
            </button>
          </div>
          <div
            class="sc-ysclju-3 EYlFI"
          >
            <span
              class="sc-1ntgl3l-0 fwKzLl"
              textalign="initial"
              textcolor="inherit"
              textvariant="body3"
            >
              100
               views
            </span>
            <button
              class="sc-1k2druj-0 eLgfbc"
              edges="square"
              type="button"
              variant="solid"
            >
              <span
                class="sc-1ntgl3l-0 eOsHHD"
                textalign="initial"
                textcolor="inherit"
                textvariant="button1"
              >
                Watch Now
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        alignitems="left"
        backgroundcolor="#ffffff"
        borderradius="0.8rem"
        class="sc-gr411i-0 XbrlJ sc-groybc-0 iOwCrz"
        display="flex"
        flex="1"
        flowdirection="vertical"
        padding="[object Object]"
        separatorcolor="#98A7B7"
        shadowcolor="#b0bcc8"
        variant="solid"
      >
        <div
          class="sc-ysclju-4 cnWXyj"
        >
          <img
            alt="video"
            class="sc-ysclju-5 hJFxSB"
            src="/thumbnails/tutorial1.jpg"
          />
          <div
            class="sc-ysclju-6 mPJQr"
          />
          <div
            class="sc-ysclju-7 hLjIhT"
          >
            <svg
              class="sc-1pf3x8a-0 iEHpIU"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.927 6.82v10.36c0 .79.87 1.27 1.54.84l8.14-5.18a1 1 0 0 0 0-1.69l-8.14-5.17a.998.998 0 0 0-1.54.84z"
                fill="#000"
                fill-opacity=".54"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-ysclju-0 eRdnVw"
        >
          <h4
            class="sc-1ntgl3l-0 jPeJxZ sc-ysclju-8 jddxpz"
            data-tooltip-id="Giving Feedback the Right Way-card-label"
            textalign="initial"
            textcolor="inherit"
            textvariant="h4"
            textweight="Medium"
          >
            Giving Feedback the Right Way
          </h4>
          <button
            class="sc-1qiq5dc-0 cDBGCD sc-ysclju-10 birZqq"
            color="#F2A14226"
            textcolor="#C0741A"
            variant="solid"
          >
            <span
              class="sc-1ntgl3l-0 hGagSU"
              textalign="initial"
              textcolor="inherit"
            >
              360-degree Feedback
            </span>
          </button>
          <span
            class="sc-1ntgl3l-0 hGagSU sc-ysclju-9 VkrDH"
            data-tooltip-id="Giving Feedback the Right Way-card-description"
            textalign="initial"
            textcolor="inherit"
            textweight="Regular"
          >
            Leave impactful feedback
          </span>
          <div
            class="sc-groybc-2 fQSnmt"
            gap="0"
          >
            <button
              class="sc-1qiq5dc-0 hmnFUL sc-ysclju-10 bmgBhz"
              color="#F3F4F6"
              variant="solid"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU"
                textalign="initial"
                textcolor="inherit"
              >
                #Peer Feedback
              </span>
            </button>
          </div>
          <div
            class="sc-ysclju-3 EYlFI"
          >
            <span
              class="sc-1ntgl3l-0 fwKzLl"
              textalign="initial"
              textcolor="inherit"
              textvariant="body3"
            >
              100
               views
            </span>
            <button
              class="sc-1k2druj-0 eLgfbc"
              edges="square"
              type="button"
              variant="solid"
            >
              <span
                class="sc-1ntgl3l-0 eOsHHD"
                textalign="initial"
                textcolor="inherit"
                textvariant="button1"
              >
                Watch Now
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
